pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -288,6 +268,10 @@ func getListenersByLoadBalancerID(client *gophercloud.ServiceClient, id string)
 		return nil, err
 	}
 
+	if len(existingListeners) == 0 {
```

Comment:
This function is intended to return a (possibly empty) list of results - ie: I think it's up to the caller to decide if len() == 0 is an error.